course-mw/course-actors-common/src/main/java/org/sunbird/learner/actors/course/CourseManagementActor.java

```diff
@@ -122,11 +122,42 @@ private void createCourse(Request request) throws Exception {
                 throw ex;
             } else {
                 throw new ProjectCommonException(
-                        ResponseCode.CLIENT_ERROR.getErrorCode(),
-                        ResponseCode.CLIENT_ERROR.getErrorMessage(),
-                        SERVER_ERROR.getResponseCode());
+                        ResponseCode.SERVER_ERROR.getErrorCode(),
+                        ResponseCode.SERVER_ERROR.getErrorMessage(),
+                        ResponseCode.SERVER_ERROR.getResponseCode());
             }
         }
     }
 
+    private void handleHierarchyData(Request request, String identifier, Map<String, String> headers) throws Exception {
+        if (request.getRequest().containsKey(SunbirdKey.HIERARCHY)) {
+            String url = getConfigValue(EKSTEP_BASE_URL) + "/content/v3/hierarchy/update";
+            HttpResponse<String> updateResponse =
+                    Unirest.patch(url)
+                            .headers(headers)
+                            .body(mapper.writeValueAsString(new HierarchyGenerationHelper().generateUpdateHierarchyRequest(request, identifier)))
```

Comment:
HierarchyGenerationHelper - Create a Global variable and use it. Don't create a new variable for each request.